core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -677,11 +678,70 @@ public boolean dropCatalog(NameIdentifier ident, boolean force)
     } catch (NoSuchMetalakeException | NoSuchCatalogException ignored) {
       return false;
 
-    } catch (IOException e) {
+    } catch (GravitinoRuntimeException e) {
+      throw e;
+
+    } catch (Exception e) {
```

Comment:
Why do we treat them differently?

Reply:
Because if we wrap the `GravitinoRuntimeException` to `RuntimeException `, the error response will be aways `INTERNAL_SERVER_ERROR`, the `CatalogExceptionHandler` can not convert it to correspond error response